lib/friendly_id/slugged.rb

```diff
@@ -300,7 +302,14 @@ def should_generate_new_friendly_id?
     end
 
     def resolve_friendly_id_conflict(candidates)
-      [candidates.first, SecureRandom.uuid].compact.join(friendly_id_config.sequence_separator)
+      candidate = candidates.first
+      uid = SecureRandom.uuid
+      if candidate && friendly_id_config.slug_limit
+        max_candidate_size = friendly_id_config.slug_limit - uid.size - friendly_id_config.sequence_separator.size
+        max_candidate_size = [max_candidate_size, 0].max
+        candidate = candidate[0...max_candidate_size]
+      end
+      [candidate, uid].compact.join(friendly_id_config.sequence_separator)
```

Comment:
There may even be a simpler version waiting to be found, but I wanted to communicate my thoughts with Ruby code. 👍